components/automate-cli/cmd/chef-automate/applications.go

```diff
@@ -46,6 +73,65 @@ func newApplicationsDisableCmd() *cobra.Command {
 	}
 }
 
+func runApplicationsListDisconnectedServicesCmd(*cobra.Command, []string) error {
+	if thresholdMinutes <= 0 {
+		return status.Errorf(status.InvalidCommandArgsError,
+			"%d is not a valid threshold time in minutes. The expected time must be greater than zero.",
+			thresholdMinutes,
+		)
+	}
+
+	if format != "json" && format != "pretty" {
+		return status.Errorf(status.InvalidCommandArgsError,
+			"%s is not a valid format type. Available formats are 'json' and 'pretty'.",
+			format,
+		)
+	}
+
+	var (
+		ctx            = context.Background()
+		apiClient, err = apiclient.OpenConnection(ctx)
+	)
+	if err != nil {
+		return status.Wrap(err, status.APIUnreachableError,
+			"Failed to create a connection to the API")
+	}
+
+	servicesRes, err := apiClient.ApplicationsClient().GetDisconnectedServices(ctx,
+		&applications.DisconnectedServicesReq{
+			ThresholdMinutes: int32(thresholdMinutes),
+		},
+	)
+	if err != nil {
+		return err
+	}
+
+	if len(servicesRes.Services) == 0 {
+		writer.Printf(
+			"There are no disconnected services with a threshold of %d minute(s)\n",
+			thresholdMinutes,
+		)
+		return nil
+	}
+
+	switch format {
+	case "json":
+		json, err := (&jsonpb.Marshaler{OrigName: true}).MarshalToString(servicesRes)
```

Comment:
Would we perhaps want `EmitDefaults` here, too, so that zero counts (`0`) would be rendered as well? (Just a suspicion, haven't tried myself.)

Reply:
Sounds like a good idea.

Reply:
Now all fields will be displayed:
```
[70][default:/src:0]# /bin/chef-automate applications list-disconnected-services --threshold-minutes 60 -f json | jq .
{
  "services": [
    {
      "supervisor_id": "d4869e97-6fd8-4069-9487-1da17ee88a3d",
      "release": "core/app/0.1.0/20101212010101",
      "group": "app.default",
      "health_check": "OK",
      "status": "RUNNING",
      "application": "demo",
      "environment": "demo",
      "fqdn": "app.example.com",
      "channel": "",
      "update_strategy": "",
      "site": "eu",
      "previous_health_check": "NONE",
      "current_health_since": "5 hours 18 minutes",
      "health_updated_at": "2019-07-11T08:13:11.884471Z"
    },
    {
      "supervisor_id": "a24632dd-61fe-4a31-96e5-efb461a8f64a",
      "release": "core/app/0.1.0/20101212010101",
      "group": "app.default",
      "health_check": "OK",
      "status": "RUNNING",
      "application": "demo",
      "environment": "demo",
      "fqdn": "app.example.com",
      "channel": "",
      "update_strategy": "",
      "site": "eu",
      "previous_health_check": "NONE",
      "current_health_since": "6 hours 3 minutes",
      "health_updated_at": "2019-07-11T07:27:50.033585Z"
    }
  ]
}
```

Reply:
I suppose it depends on your data. Stuff that will be surprising are:
- booleans (`false` is the default, so a field with value `false` would not be output)
- numbers (fields with value `0`)
- arrays (fields with empty array values)
- enums (fields with the _first_ enum value are not displayed)

the downside is visibile, too, there:
```
      "channel": "",
      "update_strategy": "",
```